src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollListBoxAutomationPeer.cs

```diff
@@ -11,19 +12,22 @@ internal class InfiniteScrollListBoxAutomationPeer : ListBoxAutomationPeer
 
         protected override List<AutomationPeer> GetChildrenCore()
         {
-            var infiniteScrollListBox = Owner as InfiniteScrollListBox;
-
-            try
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
             {
-                infiniteScrollListBox?.ItemsLock.Wait();
+                var infiniteScrollListBox = Owner as InfiniteScrollListBox;
 
-                // Don't return the LoadingStatusIndicator as an AutomationPeer, otherwise narrator will report it as an item in the list of packages, even when not visible
-                return base.GetChildrenCore()?.Where(lbiap => !(((ListBoxItemAutomationPeer)lbiap).Item is LoadingStatusIndicator)).ToList() ?? new List<AutomationPeer>();
-            }
-            finally
-            {
-                infiniteScrollListBox?.ItemsLock.Release();
-            }
+                try
+                {
+                    await infiniteScrollListBox?.ItemsLock.WaitAsync();
```

Comment:
Your `?.` operator suggests you expect `infiniteScrollListBox` to be null sometimes. If it ever were null, instead of awaiting a `Task` you would be awaiting null, which causes your method to throw `NullReferenceException`. You'll want to check for null on that first, or add `?? TplExtensions.CompletedTask` at the end of your awaited expression so you never await null.
We actually have an analyzer to catch this bug [on our backlog](https://github.com/Microsoft/vs-threading/issues/298).